### PR TITLE
Add instructions on how to export to Modrinth

### PR DESCRIPTION
**Pull request by cassiancc**
As Modrinth exporting has some unique differences to CurseForge exporting, especially where it comes to handling of optional mods, I think the website would benefit from a page covering these details. 

Happy to make any changes myself or accept any changes to the potential page.

### CODE REVIEW

File: docs/tutorials/hosting/modrinth.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Publishing to Modrinth
+
+Exporting a Modrinth pack is as simple as running `packwiz modrinth export` - this gives you a `.mrpack` in your pack folder that you can upload to Modrinth!
+
+Since this pack format doesn't support side-only mods, packwiz can't create a pack that differs between server and client, and will export a pack for Minecraft clients (containing mods with side `client` or `both`). Mods without the necessary Modrinth metadata (such as those installed from CurseForge) will be placed as JARs into the modpack zip; make sure that you have the licenses for these mods [as it is your responsibility as a pack creator to](https://support.modrinth.com/en/articles/8797527-obtaining-modpack-permissions).
```
Comment (comp500) [inline docs/tutorials/hosting/modrinth.md]: The Modrinth pack format does support side-only mods! You only need to export a pack once and the launcher will filter out server-only mods, and you can use a tool like [mrpack-install](https://support.modrinth.com/en/articles/8802250-modpacks-on-modrinth) to install the mrpack file on a server.

File: docs/tutorials/hosting/modrinth.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+Be wary of including files that you don't want (the `packwiz` executable, and the modpack zip itself) in the pack!
+
+The Modrinth pack format doesn't really support optional mods. The user won't be prompted for optional mods, and unlike with CurseForge, they'll be included in the modpack file (even if they default to being disabled!)
```
Comment (comp500) [inline docs/tutorials/hosting/modrinth.md]: The pack format itself does support optional mods, but the Modrinth app doesn't prompt the user whether to disable them. Prism Launcher (and possibly also ATLauncher) does have a prompt when installing for which optional mods should be enabled!

**Review by comp500 — CHANGES_REQUESTED** [review]

File: docs/tutorials/hosting/modrinth.md (hunk 2 above)
Comment (cassiancc) [reply docs/tutorials/hosting/modrinth.md]: Good to know! I was using Prism's official Modrinth download with my packs, which unfortunately doesn't prompt for optional mods. I've clarified that in my descirption.

File: docs/tutorials/hosting/modrinth.md (hunk 1 above)
Comment (cassiancc) [reply docs/tutorials/hosting/modrinth.md]: Thanks for clarifying that! I've updated the description to better reflect that.

**Review by comp500 — APPROVED** [review]
Thanks for the PR!

**Pull request merged**